augur/export_v2.py

```diff
@@ -857,7 +873,31 @@ def register_parser(parent_subparsers):
     )
     optional_settings.add_argument('--minify-json', action="store_true", help="export JSONs without indentation or line returns")
     optional_settings.add_argument('--include-root-sequence', action="store_true", help="Export an additional JSON containing the root sequence (reference sequence for vcf) used to identify mutations. The filename will follow the pattern of <OUTPUT>_root-sequence.json for a main auspice JSON of <OUTPUT>.json")
-    optional_settings.add_argument('--skip-validation', action="store_true", help="skip validation of input/output files. Use at your own risk!")
+    optional_settings.add_argument(
+        '--validation-mode',
+        dest="validation_mode",
+        type=ValidationMode,
+        choices=[mode for mode in ValidationMode],
+        default=ValidationMode.ERROR,
+        help="""
+            Control if optional validation checks are performed and what
+            happens if they fail.
+
+            'error' and 'warn' modes perform validation and emit messages about
+            failed validation checks.  'error' mode causes a non-zero exit
+            status if any validation checks failed, while 'warn' does not.
+
+            'skip' mode performs no validation.
+
+            Note that some validation checks are non-optional and as such are
+            not affected by this setting.
+        """)
+    optional_settings.add_argument(
+        '--skip-validation',
+        dest="validation_mode",
```

Comment:
Should we guard/warn against using both `--validation-mode` and `--skip-validation` options? 

Currently, the last provided option "wins" so 
`augur export v2 --validation-mode=error --skip-validation` will skip the validation but
`augur export v2 --skip-validation --validation-mode=error` will error when validation fails.

Reply:
I don't think so. This behaviour is consistent with general option handling when an option is repeated and only takes a single value. For example:

```
--validation-mode=error --validation-mode=skip
```

is not invalid and gets you `skip`.

This is considered a feature, as it allows hardcoded defaults (e.g. in a workflow or script) to be overridden by user/environment-provided extra args (e.g. from a workflow config).

Reply:
> This is considered a feature, as it allows hardcoded defaults (e.g. in a workflow or script) to be overridden by user/environment-provided extra args (e.g. from a workflow config).

Ahh, yeah, that makes sense!